builtin/providers/google/resource_compute_instance_group.go

```diff
@@ -0,0 +1,301 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceComputeInstanceGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeInstanceGroupCreate,
+		Read:   resourceComputeInstanceGroupRead,
+		Update: resourceComputeInstanceGroupUpdate,
+		Delete: resourceComputeInstanceGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"fingerprint": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"named_port": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"port": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"instances": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"network": &schema.Schema{
```

Comment:
This is strange, the library docs list it as `[Output Only]` see https://godoc.org/google.golang.org/api/compute/v1#InstanceGroup

From my testing the adding of instances will populate the network into the state. or leave it blank should no instance be defined when it's created.

If you later add instance the network is populated correctly. However the network remains even after all instances are removed, this could be something to change?